lib/chef/provider/remote_file/ftp.rb

```diff
@@ -148,7 +148,7 @@ def get
 
         #adapted from buildr/lib/buildr/core/transports.rb via chef/rest/rest_client.rb
         def proxy_uri(uri)
-          proxy = Chef::Config["ftp_proxy"]
+          proxy = Chef::Config["ftp_proxy"] || ENV["ftp_proxy"] || ENV["FTP_PROXY"]
```

Comment:
yeah, i'd really like to see this added to Chef::Config to pull the config variables out of the environment by default.

needs to be applied to http_proxy and https_proxy and the rest of the proxy universe as well.  it'll fix a whole large class of this kind of bug, and ensure we don't make mistakes like this going forwards.